sdk/core/azure-core/azure/core/messaging.py

```diff
@@ -0,0 +1,148 @@
+     as data_base64 in the outgoing request.
+    :type data: object
+    :keyword time: Optional. The time (in UTC) the event was generated.
+    :type time: ~datetime.datetime
+    :keyword dataschema: Optional. Identifies the schema that data adheres to.
+    :type dataschema: str
+    :keyword datacontenttype: Optional. Content type of data value.
+    :type datacontenttype: str
+    :keyword subject: Optional. This describes the subject of the event in the context of the event producer
+     (identified by source).
+    :type subject: str
+    :keyword specversion: Optional. The version of the CloudEvent spec. Defaults to "1.0"
+    :type specversion: str
+    :keyword id: Optional. An identifier for the event. The combination of id and source must be
+     unique for each distinct event. If not provided, a random UUID will be generated and used.
+    :type id: Optional[str]
+    :keyword extensions: Optional. A CloudEvent MAY include any number of additional context attributes
+     with distinct names represented as key - value pairs. Each extension must be alphanumeric, lower cased
+     and must not exceed the length of 20 characters.
+    :type extensions: Optional[Dict]
+    :ivar source: Identifies the context in which an event happened. The combination of id and source must
+     be unique for each distinct event. If publishing to a domain topic, source must be the domain name.
+    :vartype source: str
+    :ivar data: Event data specific to the event type.
+    :vartype data: object
+    :ivar type: Type of event related to the originating occurrence.
+    :vartype type: str
+    :ivar time: The time (in UTC) the event was generated.
+    :vartype time: ~datetime.datetime
+    :ivar dataschema: Identifies the schema that data adheres to.
+    :vartype dataschema: str
+    :ivar datacontenttype: Content type of data value.
+    :vartype datacontenttype: str
+    :ivar subject: This describes the subject of the event in the context of the event producer
+     (identified by source).
+    :vartype subject: str
+    :ivar specversion: Optional. The version of the CloudEvent spec. Defaults to "1.0"
+    :vartype specversion: str
+    :ivar id: An identifier for the event. The combination of id and source must be
+     unique for each distinct event. If not provided, a random UUID will be generated and used.
+    :vartype id: str
+    :ivar extensions: A CloudEvent MAY include any number of additional context attributes
+     with distinct names represented as key - value pairs. Each extension must be alphanumeric, lower cased
+     and must not exceed the length of 20 characters.
+    :vartype extensions: dict
+    """
+
+    def __init__(self, source, type, **kwargs):  # pylint: disable=redefined-builtin
+        # type: (str, str, **Any) -> None
+        self.source = source
+        self.type = type
+        self.specversion = kwargs.pop("specversion", "1.0")
```

Comment:
We are missing type annotations for all instance variables that we `pop` from `kwargs` (since they are typed as `**Any`). We should add them or else static type checkers (e.g. MyPy) will not be able to reason about the type...